### ISSUE TITLE
Bad initial guess for invIncompleteBeta 

### ISSUE THREAD

**Issue by Shimuuar** [bos/math-functions#35]
`invIncompleteBeta 7 0.07 2.865e-7` fails to converge in 10 iterations and needs 39 to get correct answer. Approximation used:

```
      -- Otherwise we revert to approximation from AS64 derived from
      -- [AS64 2] when it's applicable.
      --
      -- It slightly reduces average number of iterations when `a' and
      -- `b' have different magnitudes.
      | chi2 > 0 && ratio > 1 = 1 - 2 / (ratio + 1)
```

**Comment by Shimuuar** [comment]
More thorough investigation shows that convergence problems are quite frequent and pervasive.

For a>1 and b>1 Carter approximation is used. Overall it behaves well but for big a and b≈1 it behaves poorly. Here is plot for maximum number of iterations needed for convergence for a,b:

![download](https://cloud.githubusercontent.com/assets/100394/16387017/152fd964-3ca4-11e6-9493-f19f77d146f5.png)

![download](https://cloud.githubusercontent.com/assets/100394/16387026/294352aa-3ca4-11e6-9a95-e59f713b2d59.png)

Otherwise approximations from AS64 and numerical recipes are used. Both have areas where initial guess is very poor and large number of iterations is needed.

**Comment by Shimuuar** [comment]
Fixed in 8214bc3ddbb1ae01830f22d876409e9e662632c4,  4d8e69797304ea8620a5659331d6ed915fc94ceb, 378d6702999d4259fa1acea6b69b368f5e506bb5
